frontend/src/App.tsx

```diff
@@ -1,42 +1,79 @@
-import React, { useState } from 'react';
+import React, { useState, FC } from 'react';
 import axios from 'axios';
 import { NavBar, Indicator, PromptBox } from './components';
 
-export const App = () => {
-  const [responseFromContent, setResponseFromContent] = useState('');
-  const [isTyping, setIsTyping] = useState(false); // new state to track user typing
+interface LeetData {
+  description: string;
+  code: string;
+}
+
+const App: FC = () => {
+  const [hint, setHint] = useState<string>('');
+  const [isTyping, setIsTyping] = useState<boolean>(false);
 
   const message = {
-    from: 'Sender.React',
-    type: 'SCRAPE_HINT_DATA',
+    from: "Sender.React",
+    type: "SCRAPE_HINT_DATA",
   };
 
   const queryInfo = {
     active: true,
-    currentWindow: true,
+    currentWindow: true
+  };
+
+  const getCurrentTabId = (): Promise<number> => {
+    return new Promise((resolve, reject) => {
+      chrome.tabs.query(queryInfo, tabs => {
+        if (tabs && tabs[0] && typeof tabs[0].id === 'number') {
+          resolve(tabs[0].id);
+        } else {
+          reject(new Error('No tabs found'));
+        }
+      });
+    });
   };
 
-  const requestHint = () => {
-    chrome.tabs &&
-      chrome.tabs.query(queryInfo, (tabs) => {
-        const currentTabId = tabs[0].id;
-        if (!currentTabId) return;
-        chrome.tabs.sendMessage(currentTabId, message, (response) => {
-          setResponseFromContent(response.problem + '\n' + response.code);
-        });
+  const sendMessageToContentScript = (tabId: number): Promise<LeetData> => {
+    return new Promise((resolve) => {
+      chrome.tabs.sendMessage(tabId, message, (response) => {
+        resolve(response);
       });
+    });
+  };
+
+  const requestHint = async () => {
+    try {
+      const currentTabId = await getCurrentTabId();
+      const leetData = await sendMessageToContentScript(currentTabId);
+
+      console.log(leetData);
+
+      let hint = 'We failed to make a request to the server. Please try again.'; // default hint
+
+      const response = await axios.post('http://localhost:8000/hint', {
```

Comment:
I believe we need to add a line like `"proxy": "http://127.0.0.1: 8000"` to our `Package.json`. Then we can just set this to `/hint`

Reply:
Great idea

Reply:
Played around with it for a bit. Seems like since the project was made with Vite theres a different way to go about it. Will make a small issue for it.